dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/execute/AsyncTaskExecutionContext.java

```diff
@@ -59,19 +59,17 @@ public AsyncTaskExecutionContext(@NonNull TaskExecutionContext taskExecutionCont
         this.executeInterval = Math.max(asyncTaskExecuteFunction.getAsyncTaskStateCheckInterval().toMillis(), 1000L);
     }
 
-    public void refreshStartTime() {
-        if (executeTimes == 0) {
+    public synchronized void refreshStartTime() {
+        if (executeTimes != 0) {
             // The first time doesn't have delay
-            executeTimes++;
-        } else {
             currentStartTime = System.currentTimeMillis();
         }
+        executeTimes++;
```

Comment:
The executeTimes is only a mark flag, it is not used to represent execute the real times.

Reply:
we can record the execute times for monitoring

Reply:
yes, it's great, we can add this in task metrics.